pandas/tests/io/test_stata.py

```diff
@@ -1922,11 +1944,11 @@ def test_chunked_categorical(version):
     df.index.name = "index"
     with tm.ensure_clean() as path:
         df.to_stata(path, version=version)
-        reader = StataReader(path, chunksize=2, order_categoricals=False)
-        for i, block in enumerate(reader):
-            block = block.set_index("index")
-            assert "cats" in block
-            tm.assert_series_equal(block.cats, df.cats.iloc[2 * i : 2 * (i + 1)])
+        with StataReader(path, chunksize=2, order_categoricals=False) as reader:
```

Comment:
The iterator mode is already self-closing.  Why is the context manager introduced here?

https://github.com/pandas-dev/pandas/blob/2402abef4900385f8ee5b12a54019068dbc7d1ed/pandas/io/stata.py#L1704-L1710

Reply:
Sure, in _this_ test we're reading the reader through, but if we weren't, then it would be good form to close the reader (and the underlying file handle), just like with any file handles in Python.

In fact, you might argue the `self.close()` there in the reader is inappropriate, since an esoteric user could have e.g. saved the reader's underlying file handle's `.seek()` position at some point, would then rewind the handle and have the reader read some more... but perhaps that's an esoteric enough use case that it doesn't need to be covered. 😁